core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -259,6 +259,19 @@ private static Manifest loadLinkedManifest(File archive) throws IOException {
         return new PluginWrapper(pluginManager, archive, manifest, baseResourceURL,
                 createClassLoader(paths, dependencyLoader, atts), disableFile, dependencies, optionalDependencies);
     }
+    
+    /**
+     * Returns all the bundled plugin dependencies for a particular Jenkins version
+     */
+    public static List<PluginWrapper.Dependency> getPreviouslyBundledDependencies(String jenkinsVersion) {
+        List<PluginWrapper.Dependency> out = new ArrayList<>();
+        for (DetachedPlugin detached : DETACHED_LIST) {
+            if (jenkinsVersion == null || jenkinsVersion.equals("null") || new VersionNumber(jenkinsVersion).compareTo(detached.splitWhen) <= 0) {
+                out.add(new PluginWrapper.Dependency(detached.shortName + ':' + detached.requireVersion));
```

Comment:
:bug: reuse `DetachedPlugin.fix`, not only to avoid duplication but so you can honor `BREAK_CYCLES`.